homeassistant/components/goodwe/coordinator.py

```diff
@@ -0,0 +1,79 @@
+"""Update coordinator for Goodwe."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from goodwe import Inverter, InverterError, RequestFailedException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import SCAN_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GoodweUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Gather data for the energy device."""
+
+    inverter: Inverter
+    last_data: dict[str, Any]
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        inverter: Inverter,
+    ) -> None:
+        """Initialize update coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=entry.title,
+            update_interval=SCAN_INTERVAL,
+            update_method=self._async_update_data,
+        )
+        self.inverter = inverter
+        self.last_data = {}
```

Comment:
make this protected since it's not used outside the data update coordinator
```suggestion
        self._last_data: dict[str, Any] = {}
```

Reply:
Done.